extension/src/experiments/model/modify/collect.ts

```diff
@@ -8,32 +9,53 @@ export type Param = {
 }
 
 const collectFromParamsFile = (
-  acc: Param[],
+  acc: { path: string; value: Value | undefined }[],
   key: string | undefined,
   value: Value | ValueTree,
   ancestors: string[] = []
 ) => {
   const pathArray = [...ancestors, key].filter(Boolean) as string[]
 
-  if (isValueTree(value)) {
+  const isLeaf = !isValueTree(value)
+
+  const path = appendColumnToPath(...pathArray)
+
+  if (pathArray.length > 1) {
+    acc.push({ path, value: isLeaf ? value : undefined })
+  }
+
+  if (!isLeaf) {
     for (const [childKey, childValue] of Object.entries(value)) {
       collectFromParamsFile(acc, childKey, childValue, pathArray)
     }
-    return
   }
+}
 
-  const path = appendColumnToPath(...pathArray)
+const removeFilePrefixIfOnlyDefault = (
+  acc: Param[],
+  paramsFiles: string[]
+): Param[] => {
+  const defaultParamsFile = 'params.yaml'
+  if (paramsFiles.length === 1 && paramsFiles[0] === defaultParamsFile) {
```

Comment:
Addresses [this ask](https://github.com/iterative/vscode-dvc/pull/5030#issuecomment-1831910318).